src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -726,7 +730,9 @@ def name(self, value: Hashable) -> None:
         if not is_hashable(value):
             raise TypeError(f"{type(self).__name__}.name must be a hashable type")
         self._query_compiler = self._query_compiler.set_index_names([value])
-        if self._parent is not None:
+        # Update the name of the parent's index only if the parent's current query compiler
+        # matches the recorded query compiler (_parent_qc).
+        if self._parent is not None and self._parent_qc is self._parent._query_compiler:
```

Comment:
also now after updating, you should make sure the parent_qc is updated to the new one, you can also wrap that logic in the IndexParent class

Reply:
done!